### ISSUE TITLE
Incorrect MERGE statement syntax generated in FIREBIRD when aliasing USING derived table

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#12599]
When using the translator to translate this input query:

```sql
merge into t
using (select 1 a) s
on (1 = 1)
when matched then update set x = 1
```

Then we get the following invalid output for Firebird:

```sql
merge into T
using (
  (
    select 1 A
    from RDB$DATABASE
  )
) S
on 1 = 1
when matched then update set
  I = 1;
```

The error is:

> SQL-Fehler [335544634] [42000]: Dynamic SQL Error; SQL error code = -104; Token unknown - line 7, column 1; ) [SQLState:42000, ISC error code:335544634]

If the alias is removed:

```sql
merge into t
using (select 1 a)
on (1 = 1)
when matched then update set x = 1
```

Then the output is correct:

```sql
merge into T
using (
  select 1 A
  from RDB$DATABASE
)
on 1 = 1
when matched then update set
  i = 1;
```

The limitation of syntax in Firebird seems to be quite arbitrary, yet, other dialects might be affected as well.

**Comment by lukaseder** [comment]
The usual candidate for similar `MERGE` limitations is Derby...

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.16.0
